cdap-app-fabric/src/main/java/co/cask/cdap/internal/pipeline/PluginId.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.pipeline;
+
+import java.util.Objects;
+
+/**
+ * A wrapper class around plugin name and type which can uniquely identify a plugin.
```

Comment:
this actually is not true. An ArtifactId, plugin type, and plugin name unique identifies a plugin.